tests/framework/base/ComponentTest.php

```diff
@@ -376,6 +380,9 @@ public function testDetachBehavior()
 
         $detachedBehavior = $component->detachBehavior('z');
         $this->assertNull($detachedBehavior);
+
+        $this->expectException('yii\base\UnknownMethodException');
+        $component->test();
```

Comment:
This made more sense to test here than in `attachBehaviors` where it lived and was causing the test to end prematurely.